services/app-api/prisma/schema.prisma

```diff
@@ -175,62 +175,47 @@ model SharedRateCertifications {
-  updatedAt          DateTime               @default(now()) @updatedAt
-  name               String?
-  titleRole          String?
-  email              String?
-  actuarialFirm      ActuarialFirm?
-  actuarialFirmOther String?
-  rateWithCertifyingActuaryID    String?
-  rateWithAddtlActuaryID    String?
-  rateActuaryCertifying  RateRevisionTable?  @relation(name: "CertifyingActuaryOnRateRevision", fields: [  rateWithCertifyingActuaryID ], references: [id])
-  rateActuaryAddtl   RateRevisionTable?   @relation(name: "AddtlActuaryOnRateRevision", fields: [   rateWithAddtlActuaryID ], references: [id])
+  id                          String             @id @default(uuid())
+  createdAt                   DateTime           @default(now())
+  updatedAt                   DateTime           @default(now()) @updatedAt
+  name                        String?
+  titleRole                   String?
+  email                       String?
+  actuarialFirm               ActuarialFirm?
+  actuarialFirmOther          String?
+  rateWithCertifyingActuaryID String?
+  rateWithAddtlActuaryID      String?
+  rateActuaryCertifying       RateRevisionTable? @relation(name: "CertifyingActuaryOnRateRevision", fields: [rateWithCertifyingActuaryID], references: [id])
+  rateActuaryAddtl            RateRevisionTable? @relation(name: "AddtlActuaryOnRateRevision", fields: [rateWithAddtlActuaryID], references: [id])
+  display_seq                 Int?
 }
 
 model ContractDocument {
-  id                 String                @id @default(uuid())
-  createdAt          DateTime              @default(now())
-  updatedAt          DateTime              @default(now()) @updatedAt
-  name               String
-  s3URL              String
-  sha256             String
-  contractRevisionID String
-  contractRevision   ContractRevisionTable @relation(fields: [contractRevisionID], references: [id])
-}
-
-model ContractSupportingDocument {
-  id                 String                @id @default(uuid())
-  createdAt          DateTime              @default(now())
-  updatedAt          DateTime              @default(now()) @updatedAt
-  name               String
-  s3URL              String
-  sha256             String
-  contractRevisionID String
-  contractRevision   ContractRevisionTable @relation(fields: [contractRevisionID], references: [id])
+  id                                 String                 @id @default(uuid())
+  createdAt                          DateTime               @default(now())
+  updatedAt                          DateTime               @default(now()) @updatedAt
+  name                               String
+  s3URL                              String
+  sha256                             String
+  contractDocumentRevisionID         String
+  supportingDocumentRevisionID       String
+  contractDocumentContractRevision   ContractRevisionTable? @relation(name: "ContractDocumentOnContractRevision", fields: [contractDocumentRevisionID], references: [id])
```

Comment:
If we're going to do it this way, why not have a single documents table? 

The reason I suggested we have 4 different tables here was that then the primary key could be required in the database, ensuring that we don't accidentally orphan a document at some point. But I don't think that's that big a deal, it would be nice if we could add a db constraint that made sure that one of the four primary keys were set so you can't make a document that doesn't belong to anything.

Reply:
@macrael It could be a single table but in that case we lost a document having to have a relationship with either a contract or a rate revision would not be a requirement. Right now because they are separate a RateDocument has a required relationship to its rate revision, similar with contract docs and contract. Do we want to lose that granularity?  I believe the revisionID relationship has to then be optional if its all one